pep-0723.rst

```diff
@@ -0,0 +1,185 @@
+ecosystems (namely `Rust <https://github.com/rust-lang/rfcs/blob/master/text/3424-cargo-script.md>`__
+and `Go <https://github.com/erning/gorun>`__) by embedding the existing
+standard :pep:`621` that is used to describe projects.
+
+Any Python script may assign a variable named ``__pyproject__`` to a multi-line
+double-quoted string containing a valid TOML document. This document contains
+what you would expect to see in a ``pyproject.toml`` file, with the following
+exceptions regarding the ``[project]`` table:
+
+* The ``name`` and ``version`` fields MUST be defined dynamically by tools if
+  the user does not define them
+* These fields do not need to be listed in the ``dynamic`` array
+
+Example
+-------
+
+The following is an example of a script with an embedded ``pyproject.toml``:
+
+.. code:: python
+
+    __pyproject__ = """
+    [project]
+    requires-python = ">=3.11"
+    dependencies = [
+      "requests<3",
+      "rich",
+    ]
+    """
+
+    import requests
+    from rich.pretty import pprint
+
+    resp = requests.get("https://peps.python.org/api/peps.json")
+    data = resp.json()
+    pprint([(k, v["title"]) for k, v in data.items()][:10])
+
+
+Reference Implementation
+========================
+
+This regular expression may be used to parse the metadata::
+
+    (?ms)^__pyproject__ *= *"""$(.+?)^"""$
+
+For languages that do not support easily accessing the match group, one may
+parse the entire match as TOML and then access the ``__pyproject__`` key.
+
+The following is an example of how to read the metadata on Python 3.11 or
+higher.
+
+.. code:: python
+
+    import re, tomllib
+
+    def read(script: str) -> dict | None:
+        match = re.search(r'(?ms)^__pyproject__ *= *"""$(.+?)^"""$', script)
+        return tomllib.loads(match.group(1)) if match else None
+
+Often tools will provide commands to manage dependencies. The following is a
+crude example of modifying the content using the ``tomlkit`` library.
+
+.. code:: python
+
+    import re, tomlkit
+
+    def add(script: str, dependency: str) -> str:
+        match = re.search(r'(?ms)^__pyproject__ *= *"""$(.+?)^"""$', script)
+        config = tomlkit.parse(match.group(1))
+        config['project']['dependencies'].append(dependency)
+
+        start, end = match.span(1)
+        return script[:start] + tomlkit.dumps(config) + script[end:]
+
+
+Backwards Compatibility
+=======================
+
+At the time of writing, ``__pyproject__`` only appears 5 times
+`on GitHub <https://github.com/search?q=__pyproject__&type=code>`__ and 4 of
+those belong to a user who appears to already be using this PEP's format
+for its intended purpose.
+
+
+How to Teach This
+=================
+
+The format is intended to bridge the gap between users just writing scripts and
+users writing packages. Knowledge of how to write metadata for one use case will
+be directly transferable to the other.
+
+Everything in the parent PEP regarding teaching is applicable to this as well.
```

Comment:
If by "the parent PEP" you mean PEP 722 (in which case, please state the PEP number explicitly, I don't think this is a "child" of that PEP, but rather an "alternative"), then that's just flat-out wrong. Maybe you were looking at the old version, but the current version of PEP 722 says:

> The format is intended to be close to how a developer might already specify script dependencies in an explanatory comment. The required structure is deliberately minimal, and the concept of using a special comment marker (``##`` in this case) is not unusual (the "shebang" line in a Unix shell script is an example).

Also the whole "Why not use a more standard data format (e.g., TOML)?" section of PEP 722 is making the point that TOML is hard to teach and understand for the target audience, so claiming that PEP 722 supports the "how to teach" discussion in this PEP is not just inaccurate, but also misleading.